django/db/backends/base/operations.py

```diff
@@ -204,6 +204,22 @@ def for_update_sql(self, nowait=False, skip_locked=False, of=()):
             ' SKIP LOCKED' if skip_locked else '',
         )
 
+    def _prepare_limit_offset_params(self, low_mark, high_mark):
+        offset = low_mark or 0
+        if high_mark is not None:
+            return (high_mark - offset), offset
+        elif offset:
+            return self.connection.ops.no_limit_value(), offset
+        return None, offset
+
+    def limit_offset_sql(self, low_mark, high_mark):
+        """Return LIMIT/OFFSET SQL clause."""
+        limit, offset = self._prepare_limit_offset_params(low_mark, high_mark)
+        return '%s%s' % (
+            ' LIMIT %d' % limit if limit else '',
```

Comment:
It would clear ambiguity if you added some parentheses here and on the next line: `(' LIMIT %d' % limit) if limit ...`. Initially, I thought the grouping was `% (limit if limit else '')`.